src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -275,4 +279,56 @@ public static AssemblyResultSet assembleReads(final AssemblyRegion region,
             throw e;
         }
     }
+
+    /**
+     * Annotates reads in ReadLikelihoods with alignment region (the ref region spanned by the haplotype the read is aligned to) and
+     * callable region (the ref region over which a caller is using these ReadLikelihoods to call variants)
+     *
+     * @param likelihoodsHaplotype ReadLikelihoods containing reads to be annotated along with haplotypes to which these reads have been aligned
+     * @param callableRegion ref region over which caller is using these ReadLikelihoods to call variants
+     */
+    public static void annotateReadLikelihoodsWithRegions(final ReadLikelihoods<Haplotype> likelihoodsHaplotype,
+                                                          final Locatable callableRegion) {
+        //assign alignment regions to each read
+        final Collection<ReadLikelihoods<Haplotype>.BestAllele> bestHaplotypes = likelihoodsHaplotype.bestAllelesBreakingTies();
+        for (final ReadLikelihoods<Haplotype>.BestAllele bestHaplotype : bestHaplotypes) {
+            final GATKRead read = bestHaplotype.read;
+            final Haplotype haplotype = bestHaplotype.allele;
+            read.setAttribute(ALIGNMENT_REGION_TAG, haplotype.getGenomeLocation().toString());
+        }
+
+        //assign callable region to each read
+        final int sampleCount = likelihoodsHaplotype.numberOfSamples();
+        for (int i = 0; i < sampleCount; i++) {
+            for (final GATKRead read : likelihoodsHaplotype.sampleReads(i)) {
+                read.setAttribute(CALLABLE_REGION_TAG, callableRegion.toString());
+            }
+        }
+    }
+
+    /**
+     * Annotates reads in ReadLiklihoods with genotype supported by each read a particular variant.  If a read already has a
```

Comment:
"Annotates reads in ReadLikelihoods with genotype supported by each read at a particular variant." ?